### PR TITLE
TST: Add new tests to ensure proper pip install of RC

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

In several ecosystem packages, pysat 3.2.0 does not properly install to run tests for the operational environment, and the command to set the data directory fails.

This pull adds all tested environs to the pip rc workflow. The initial run documents the [same failure](https://github.com/pysat/pysat/actions/runs/8471450446) in the ecosystem packages: 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Running the [manual](https://github.com/pysat/pysat/actions/workflows/pip_rc_install.yml) pip_rc_install.yml workflow in GitHub Actions.

**Test Configuration**:
* GitHub Actions

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have linted the files updated in this pull request
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes
- [x] Update zenodo.json file for new code contributors

If this is a release PR, replace the first item of the above checklist with the
release checklist on the wiki:
https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

**Comment by aburrell** [comment]
I tested this locally and found that when installing from the git repository locally, python 3.6 will install correctly:
```
python3.6 -m build .
pip-3.6 install .
python3.6 -c "import pysat; print(pysat.__version__)"
```
yields: 3.2.0

If I uninstall this and reinstall from a different directory and download pysat from PyPi it fails, but is not aware of the failure.  In the site-packages directory, there is a `pysat-3.2.0.dist-info` directory, but pysat is not installed.  The test command then (obviously) fails to import pysat.

File: .github/workflows/pip_rc_install.yml [hunk 1]
```diff
@@ -12,8 +12,21 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
-        python-version: ["3.11"]  # Keep this version at the highest supported Python version
+        os: ["ubuntu-latest", "windows-latest", "macos-latest"]
+        python-version: ["3.10", "3.11", "3.12"]
```
Comment (jklenzing) [inline .github/workflows/pip_rc_install.yml:16]:
```suggestion
        python-version: ["3.11", "3.12"]
```

File: .github/workflows/pip_rc_install.yml [hunk 2]
```diff
@@ -12,8 +12,21 @@ jobs:
...
+        numpy_ver: ["latest"]
+        test_config: ["latest"]
+        include:
+          # NEP29 compliance settings
+          - python-version: "3.9"
```
Comment (jklenzing) [inline .github/workflows/pip_rc_install.yml:21]:
```suggestion
          - python-version: "3.10"
```

**Comment by aburrell** [comment]
I think ops will work if we:
- [ ] re-add the setup.py file
- [ ] use pip version 18.0 (less than 19.0 at least)

I tried in the virtual env and got "need setup.py" with 18.0 and got the silent failure with 19.0.

**Comment by jklenzing** [comment]
> I think ops will work if we:
> 
> * [ ]  re-add the setup.py file
> * [ ]  use pip version 18.0 (less than 19.0 at least)
> 
> I tried in the virtual env and got "need setup.py" with 18.0 and got the silent failure with 19.0.

Doesn't work.

**Comment by aburrell** [comment]
Ok, then we go the "install by github only for ops" route now, as discussed yesterday, and take it out of the pip testing RC.

**Comment by jklenzing** [comment]
> Ok, then we go the "install by github only for ops" route now, as discussed yesterday, and take it out of the pip testing RC.

So, update the main workflow to install via github and remove ops testing from the RC install tests? We will still have the hole in testing where we might discover an error after release.

**Comment by aburrell** [comment]
> remove ops testing from the RC install tests

What about "install from RC branch for RC tests"?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 name: Pytest with Flake8
 
-on: [push, pull_request]
+# on: [push, pull_request]
```
Comment (jklenzing) [inline .github/workflows/main.yml:7]: Note to self: Turn back on before final review.

**Review by aburrell — APPROVED** [review]

**Comment by jklenzing** [comment]
Noting that technically versions under 3.6 still don't install properly under pip, but this is now described on the readme. The tests remain in the manual workflow for potential future improvements.

**Pull request merged**